code/game/machinery/porta_turret/portable_turret.dm

```diff
@@ -403,9 +411,18 @@
 
 			//if the target is a human and not in our faction, analyze threat level
 			if(ishuman(C) && !in_faction(C))
+				
 				if(assess_perp(C) >= 4)
-					targets += C
-
+					if(shoot_heads_of_staff)
+						targets += C
+						continue
+					else 
+						var/obj/item/card/id/I = C.get_idcard(TRUE)
+						if (I.assignment != null & I.assignment in GLOB.command_positions)

```

Comment:
If `I` is null, you can't check values associated with I, and as a result it will cause a runtime. Ergo, you'll have to check that I exists first, you can do this either as 
` 						if (I && I.assignment in GLOB.command_positions)`
or 
` 						if (I?.assignment in GLOB.command_positions)`

I'd recommend using the latter in this case as it's cleaner
(more info here; https://en.wikipedia.org/wiki/Safe_navigation_operator )

Also for future reference, AND operators are `&&`, `&` is only used for bitwise operations.